pass.go

```diff
@@ -0,0 +1,1054 @@
+	"block-all-mixed-content;",
+	"sandbox",
+	"require-sri-for script style;",
+	"base-uri 'none';",
+}, " ")
+
+var featurePolicy = strings.Join([]string{
+	"accelerometer 'none';",
+	"ambient-light-sensor 'none';",
+	"autoplay 'none';",
+	"camera 'none';",
+	"encrypted-media 'none';",
+	"fullscreen 'none';",
+	"geolocation 'none';",
+	"gyroscope 'none';",
+	"magnetometer 'none';",
+	"microphone 'none';",
+	"midi 'none';",
+	"payment 'none';",
+	"picture-in-picture 'none';",
+	"speaker 'none';",
+	"sync-xhr 'none';",
+	"sync-script 'none';",
+	"usb 'none';",
+	"vr 'none';",
+}, " ")
+
+// generateSecurityHeaders generates the http securityHeaders to be outputted
+// as http headers before every request
+func (p *Pass) generateSecurityHeaders() {
+	p.sv.securityHeaders = make(map[string]string)
+	// HSTS
+	p.sv.securityHeaders["Strict-Transport-Security"] = fmt.Sprintf("max-age=%d", p.sv.cfg.HSTS.MaxAge)
+	if p.sv.cfg.HSTS.IncludeSubDomains {
+		p.sv.securityHeaders["Strict-Transport-Security"] += "; includeSubDomains"
+	}
+	if p.sv.cfg.HSTS.Preload {
+		p.sv.securityHeaders["Strict-Transport-Security"] += "; preload"
+	}
+
+	p.sv.securityHeaders["Access-Control-Allow-Origin"] = p.GetURL()
+	p.sv.securityHeaders["Feature-Policy"] = featurePolicy
+	p.sv.securityHeaders["Referrer-Policy"] = "no-referrer"
+	p.sv.securityHeaders["X-Content-Type-Options"] = "nosniff"
+	p.sv.securityHeaders["X-Frame-Options"] = "deny"
+	p.sv.securityHeaders["X-XSS-Protection"] = "1; mode=block"
+}
+
+// serverError takes in any error this program is designed to handle.
+//
+// It returns if the error is a server side problem, or something the user sent
+// that was malformed, and the string to be displayed to the user, unless we
+// choose not to display it for obscurity.
+func (p *Pass) serverError(err error) (statusCode int, t *template.Template, txt string) {
+	if passErr, okPassErr := err.(passError); okPassErr {
+		if pqErr, okPqErr := passErr.error.(*pq.Error); okPqErr {
+			log.Printf("pqErr: %s (Error at: %v)", passErr.location, passErr)
+
+			if pqErr.Routine == "ClientAuthentication" {
+				return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+			}
+
+			return http.StatusInternalServerError, p.sv.templates.Errors500, "You caused an unknown database error... good job..."
+		} else if netErr, okNetErr := passErr.error.(net.Error); okNetErr {
+			if opErr, okOpErr := netErr.(*net.OpError); okOpErr {
+				if dnsErr, okDNSErr := opErr.Err.(*net.DNSError); okDNSErr {
+					log.Printf("DNS Lookup error, temporary: %t, timeout: %t: %v", dnsErr.IsTemporary, dnsErr.IsTimeout, dnsErr)
+					return http.StatusInternalServerError, p.sv.templates.Errors500, ""
+				} else if osSyscallErr, okOsSyscallErr := opErr.Err.(*os.SyscallError); okOsSyscallErr {
+					log.Printf("Syscall %s resulted in code %d during %s in: %v", osSyscallErr.Syscall, osSyscallErr.Err, opErr.Op, opErr)
+					return http.StatusInternalServerError, p.sv.templates.Errors500, ""
```

Comment:
Remove this line.

Reply:
Remove the return? But if i do that then the `log.Printf("net.OpError: %v", opErr)` will output a second, now redundant error message to the log.